src/main/kotlin/com/atlassian/performance/tools/jirasoftwareactions/api/page/BrowseBoardsPage.kt

```diff
@@ -28,20 +28,27 @@ class BrowseBoardsPage(
 
     fun getBoardIds(): Collection<String> =
         driver.findElements(By.cssSelector(".boards-list tr"))
-            .map { it.getAttribute("data-board-id") }
-            .filterNotNull()
+            .mapNotNull { it.getAttribute("data-board-id") }
 
     fun getScrumBoardIds(): Collection<String> {
         val boardsBeforeFiltering = driver.findElements(By.cssSelector(".boards-list tr"))
         if (boardsBeforeFiltering.isEmpty()) {
             return emptyList()
         }
-        driver.findElement(By.cssSelector("#ghx-manage-boards-filter a")).click()
-        driver.findElement(By.className("type-filter-scrum")).click()
         driver.wait(
-            Duration.ofSeconds(1),
-            ExpectedConditions.and(
-                *boardsBeforeFiltering.map { ExpectedConditions.stalenessOf(it) }.toTypedArray()
+            Duration.ofSeconds(5), 
+            elementToBeClickable(By.cssSelector("#ghx-manage-boards-filter a"))
+        ).click()
+
+        driver.wait(
+            Duration.ofSeconds(5),
+            elementToBeClickable(By.className("type-filter-scrum"))
+        ).click()
+        
+        driver.wait(
+            Duration.ofSeconds(60),
```

Comment:
It's too long of a timeout for production. `BrowseBoardsPage` is widely used in the wild.

Reply:
The reason why it's 60 here is that it should not be needed, unless it's a chromedriver problem which means it needs to be fixed in chromedriver or the condition needs to be different.